assets/product-info.js

```diff
@@ -45,57 +48,43 @@ if (!customElements.get('product-info')) {
       publish(PUB_SUB_EVENTS.quantityUpdate, undefined);  
     }
 
-    onVariantChange() {
-      this.fetchCartQty()
-    }
-
-    fetchCartQty() {
+    fetchQuantityRules() {
       this.querySelector('.quantity__rules-cart .loading-overlay').classList.remove('hidden');
-      fetch("/cart?section_id=main-cart-items")
-      .then((response) => response.text())
-      .then((responseText) => {
-        const html = new DOMParser().parseFromString(responseText, 'text/html')
-        const sourceQty = html.querySelector((`[data-quantity-variant-id~="${this.currentVariant.value}"]`))
-        const quantityRulesCartClassname = '.quantity__rules-cart';
-        if (sourceQty) {
-          const valueQtyCart = sourceQty.value;         
-          this.querySelector(quantityRulesCartClassname).classList.toggle('hidden', valueQtyCart <= 0);
-          if (valueQtyCart && this.input) {
-            this.input.dataset.cartQuantity = valueQtyCart;
-            this.destinationQty.innerHTML = valueQtyCart;
-          }
-        } else {
-          if (this.input) this.input.dataset.cartQuantity = 0;
-          this.destinationQty.innerHTML = 0;
-          this.querySelector(quantityRulesCartClassname).classList.add('hidden')
-        }
-        this.fetchQtyRules();
-      })
-      .catch(e => {
-        console.error(e);
-      });
-    }
-
-    fetchQtyRules() {
       fetch(`${this.dataset.url}?variant=${this.currentVariant.value}&section_id=${this.dataset.section}`).then((response) => {
         return response.text()
       })
       .then((responseText) => {
-        const html = new DOMParser().parseFromString(responseText, 'text/html')
-        const sourceQty = html.querySelector(('.quantity-input'))
-        const sourceQtyRules = html.querySelector(('.quantity__rules'))
-        if (sourceQty && sourceQtyRules) {
-          this.input.innerHTML = sourceQty.innerHTML
-          this.querySelector('.quantity__rules').innerHTML = sourceQtyRules.innerHTML
-        }
+        const html = new DOMParser().parseFromString(responseText, 'text/html');
+        this.updateQuantityRules(this.dataset.section, html);
         this.setQuantityBoundries();
-        // TODO: Move this to ensure loading state is removed even if error is thrown
-        this.querySelector('.quantity__rules-cart .loading-overlay').classList.add('hidden');
       })
       .catch(e => {
         console.error(e);
+      })
+      .finally(() => {
+        this.querySelector('.quantity__rules-cart .loading-overlay').classList.add('hidden');
       });
     }
+
+    updateQuantityRules(sectionId, html) {
+      const quantityFormCurrent = document.getElementById(`Quantity-Form-${sectionId}`);
+      const quantityFormUpdated = html.getElementById(`Quantity-Form-${sectionId}`);
+      const selectors = ['.quantity__input', '.quantity__rules', '.quantity__label'];
+      for (let selector of selectors) { 
+        const current = quantityFormCurrent.querySelector(selector);
```

Comment:
Variant switching is broken inside Quick Buy because of this line. Another error, on the same line, appears when you add an item to your cart.

We use `Quantity-Form-quickadd-template--17128870936598__main` for Quick Buy, while `Quantity-Form-template--17128870936598__main` is used outside. That's why we had the `data-original-section` and `data-section` in the variant switching components.

We have to change ids for Quick Add to avoid duplicates (ex.: if you open a quick Modal for **Product 1** on while you're viewing that same product page).

Reply:
Thanks for double-checking! I admittedly had not tested the quick add flow properly. Updated - https://github.com/Shopify/dawn/pull/2251/commits/1995689ba775878e6e98f95f628640a1e8f9df57